src/Model/collections.ts

```diff
@@ -152,6 +168,19 @@ Model.prototype.getOrCreateDoc = function(collectionName, id, data) {
   return collection.getOrCreateDoc(id, data);
 };
 
+Model.prototype.getOrDefault = function<S>(subpath: Path, defaultValue: S) {
+  return this.get(subpath) ?? defaultValue as ReadonlyDeep<S>;
+};
+
+Model.prototype.getOrThrow = function<S>(subpath?: Path) {
+  const value = this.get(subpath);
+  if (value === undefined) {
+    const fullpath = [this._at, subpath].filter(Boolean).join('.');
```

Comment:
Can use `this.path(subpath)` for this to not repeat logic:
https://github.com/derbyjs/racer/blob/3e0d4995dd50fa57413d1807ab5ecdfe02b97e83/src/Model/paths.ts#L27-L35